lib/provisioning/Provisioner.js

```diff
@@ -0,0 +1,112 @@
+/*eslint no-invalid-this: 0*/ // eslint doesn't understand Promise.coroutine wrapping
+"use strict";
+var Promise = require("bluebird");
+var IrcRoom = require("../models/IrcRoom");
+var MatrixRoom = require("matrix-appservice-bridge").MatrixRoom;
+
+var log = require("../logging").get("Provisioner");
+
+function Provisioner(ircBridge) {
+    this._ircBridge = ircBridge;
+
+    let as = this._ircBridge.getAppServiceBridge().appService;
+
+    as.app.post("/_matrix/provision/link", (req, res) => {
+        return this.validate(req.body).then(
+            () => { res.json({}) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        ).then(
+            () => { return this.link(req.body) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        );
+    });
+
+    as.app.post("/_matrix/provision/unlink", (req, res) => {
+        return this.validate(req.body).then(
+            () => { res.json({}) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        ).then(
+            () => { return this.unlink(req.body) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        );
+    });
+}
+
+var parameterValidation = {
+    matrix_room_id :
+        {regex : /^![0-9A-Za-z]+?:[^:]+(\:[0-9]+)?$/, example : '!Abcdefg:example.com[:8080]'},
```

Comment:
We don't currently specify a grammar for room IDs in [the spec](http://matrix.org/docs/spec/intro.html#room-ids-and-event-ids) so I would probably just check that it passes something like `!.*:.*`